arangod/VocBase/Methods/AqlUserFunctions.cpp

```diff
@@ -230,19 +230,12 @@ Result arangodb::registerUserFunction(TRI_vocbase_t* vocbase,
   bool isDeterministic = userFunction.get("isDeterministic").getBool();
   /// TODO ^^^ error handling
   
-  V8Context* v8Context = nullptr;
-  
-  ISOLATE;
-  
-  if (isolate == nullptr) {
-    v8Context = V8DealerFeature::DEALER->enterContext(vocbase, true /*allow use database*/);
-    if (!v8Context) {
-      return Result(TRI_ERROR_INTERNAL, "could not acquire v8 context while registering AQL user function");
-    }
-    isolate = v8Context->_isolate;
-  }
-
   {
+    ISOLATE;
+    V8ContextDealerGuard guard(isolate);
+    isolate = guard.isolate();
```

Comment:
I do not really like it being called with an isolate as an input parameter, and then returning an isolate (same or different) by calling a member function.